book/analysis/introduction.md

```diff
@@ -5,68 +5,53 @@ position: 1
 layout: chapter.html
 ---
 
-Every software engineer will have a different answer to the question
-“what makes one computer program better than another?” Some may focus
-on the readability of the code, others on the engineering paradigms
-used, others on “efficiency”, and so on.
-
-When we put aside the language- or implementation-specific aspects of a
-program and talk about the underlying _algorithm_, there are fewer
-comparisons to be made. We concern ourselves here with comparing
-algorithms based upon the amount of computing resources used, in other
-words we want to be able to compare two algorithms and say that one is
-better than the other because it is more economical in its use of
-computing resources.
-
-Typically, the two “computing resources” that we care about are space
-(such as memory) and time. In different contexts, we may be willing to
-compromise one more than another, for instance we may be willing for
-a program to run longer so long as it doesn’t exceed a certain amount
-of allocated space, or conversely we may be willing to use more memory
-in order to reduce the running time of a program. As such we want a
-vocabulary and method of analysis that allows us to discuss the usage
-of space and time for any algorithm we encounter.
+What makes one computer program better than another?
 
-One way we _could_ measure the running time for a program is to do a
-benchmark analysis. This would involve tracking the actual time required
-for the program to compute its result.
+The answer depends on the criteria we use: code readability, engineering
+paradigms, and efficiency are a few possible metrics by which we we could
+measure a program's quality.
 
-In Python, we can benchmark a function by noting the starting time and
-ending time with respect to the system we are using. In the `time`
-module there is a function called `time` that will return the current
-system clock time in seconds since some arbitrary starting point. By
-calling this function twice, at the beginning and at the end, and then
-computing the difference, we can get an exact number of seconds
-(fractions in most cases) for execution.
+Let's set aside aspects of a program that are specific to language or
+implementation and instead focus on the underlying _algorithm_. Doing this
+limits the number of factors we have to consider and enables us to make bold
+statements like "Tesla's algorithm is better than Edison's algorithm because
+because it uses computing resources more efficiently."
 
-Let us do so for a simple function computing the sum of a range of
-integers:
+Typically, the two “computing resources” that we care about are *space* and
+*time*. We usually replace "space" with "memory" because humans are
+narcissistic and like comparing computers to their brains.
```

Comment:
@aleksharma12 thinking about this some more, we should probably just use "space" rather than "memory" throughout the book whenever talking about algorithms in the abstract. If we are actually implementing an algorithm in a language, to be executed by an operating system process, then we can talk about memory.